### DISCUSSION TITLE
Reduce the use of macros

### DISCUSSION THREAD

**Discussion by C-BJ** [inanevin/LinaEngine#65] [Ideas]
Compilation errors in release mode, which take me so long to debug because of abusive macros
I will open a PR to repair the existing macro

**Comment by inanevin** [comment]
I'd say macros are not abused in the Lina Engine codebase, especially when compared to many projects where meta/template programming is the primary strategy.

Do you have any specific macro usage where replacing it with a function or another way is more suitable? Please give examples & points.
